sdks/python/apache_beam/runners/common.py

```diff
@@ -402,7 +402,7 @@ def __init__(self,
     self.window_fn = windowing.windowfn
 
     self.do_fn_invoker = DoFnInvoker.create_invoker(
-        do_fn_signature, context, side_inputs, args, kwargs)
+        self, do_fn_signature, context, side_inputs, args, kwargs)
```

Comment:
Turns out, memory usage significantly increases when we pass DoFnRunner to DoFnInvoker (as the output_processor) and maintain a reference to it. I suspect this is due to recursive referencing between DoFnInvoker and DoFnRunner. I could fix this by creating a new OutputProcessor class that contains the process_output method. Processing time improvement introduced by Robert's update stays the same.